### DISCUSSION TITLE
Localplay setup - permissions - pulseaudio

### DISCUSSION THREAD

**Discussion by cnic023** [ampache/ampache#3597] [Q&A]
I have set up a localplay instance with MPD -> PulseAudio. Everything works fine. I can create playlists etc BUT I can't play songs.
I get the error message in MPD log: "Failed to open "PulseAudio" (pulse; Failed to connect; Connection refused. Also: "player: problems opening audio device while playing "name of song"). 
Pulse and MPD are in the audio group - MPD is in the pulse group and the pulse-access group - MPD on localhost:6600 - Pulseaudio on localhost:8900 - the Ampache instance is on the same server.
Pulse config in mpd.conf is:
audio_output {
        type            "pulse"
        name            "PulseAudio"
        server          "localhost"             # optional
        mixer_type      "software"
##      media_role      "media_role"            #optional
}
Alsa audio_output is commented out.

I hope this is an easy fix but I can't get to the bottom of it.
I'm running Ampache  5.5.6-release - Ubuntu 22.04 server - PHP 8.0.30

**Comment by cnic023** [comment]
I tried changing audio_output to alsa so I commented out 'pulse'. 
Following from the mpd.log:
ALSA lib pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Timeout

My goal is to play my music from the ampache server via hdmi to my Pioneer vsx934 receiver. But first I want to get it playing by localplay from the server headphone jack or the audio output jack. I can probably sort the final step myself. 

PulseAudio seems to be the problem. Perhaps 'Pulse' and 'ALSA' should both be uncommented? I am a complete newb at this.

**Reply by lachlan-00** [reply]
i use moode for localplay. also has the benifits of bluetooth, spotify and local playlists. https://moodeaudio.org/

**Comment by cnic023** [comment]
Thanks for this but I'm a bit confused. Am I right in thinking I can continue to use Ampache as the GUI interface? I would also use MPD to set up the localplay instance but alter audio_output to a Moode configured output? I am running Ubuntu server 22.04 with no GUI and Ampache resides on this server but it is attached to my Pioneer receiver by HDMI. I would not use pulseaudio at all?

**Comment by cnic023** [comment]
PS - the Pioneer VSX934 is on the LAN.

**Comment by stebe** [comment]
This is my pulse localplay settings in mpd. I also have a password set in mpd. You may need to set the permissions so mpd can access the pulseaudio instance. 

```
audio_output {
        type            "pulse"
        name            "MPD Pulse"
        server          "10.2.1.5"              # optional
}

```

**Comment by cnic023** [comment]
I have fixed pulseaudio but now I think the problem is in my setup of MPD. Its log reports:
 <<exception: Failed to decode https://[domain-name]/play/ssid/318c715fef5e805fab50cfeda4ee3cff/type/song/oid/1730/uid/1/name/Gabriel%20Yared%20-%20City%20of%20Angels.mp3; CURL failed: The requested URL returned error: 404>> 
I am trying to play music from the Ubuntu 22.04 server to a Receiver <physically attached via HDMI>. Ampache and MPD are on that same server as is the music repository directory of Ampache. MPD is on 127.0.0.1:6600 - my MPD instance on Ampache is configured identically. 
If the URL in the above log snippet is received in that form from Ampache to MPD, I am confused. I would expect it to be "file://...." form or similar.  Perhaps there is a rewrite rule I should be putting into my Apache virtual host file?
It seems similar to  "Localplay: mpd receives wrong url, curl times out #1108 "

**Reply by cnic023** [reply]
Here is my mpd.conf
Perhaps I have something wrong in there?
[mpd.conf.txt](https://github.com/ampache/ampache/files/12613336/mpd.conf.txt)
